src/uu/tail/src/args.rs

```diff
@@ -182,19 +182,53 @@ impl Settings {
     }
 
     pub fn from(matches: &clap::ArgMatches) -> UResult<Self> {
-        let mut settings: Self = Self {
-            follow: if matches.get_flag(options::FOLLOW_RETRY) {
-                Some(FollowMode::Name)
-            } else if matches.value_source(options::FOLLOW) != Some(ValueSource::CommandLine) {
-                None
-            } else if matches.get_one::<String>(options::FOLLOW)
-                == Some(String::from("name")).as_ref()
+        // We're parsing --follow, -F and --retry in one go under the following conditions:
+        // * -F sets --retry and --follow=name (the -F flag is restricted by clap to be set only
+        // once in contrast to gnu's tail which allows this flag to be specified multiple times)
+        // * plain --follow or short -f is the same like specifying --follow=descriptor and can
+        // overwrite itself (specified multiple times)
+        // * specifying --retry together with -F doesn't change the setting of `retry` to true. The
+        // --retry flag itself is restricted by clap to be specified only once (unlike gnu's tail)
+        let (follow, retry) = match (
+            matches.get_flag(options::FOLLOW_RETRY),
+            matches.get_one::<String>(options::FOLLOW),
+        ) {
```

Comment:
Let's see if we can reduce the branches here. Maybe:
```rust
let arg_follow_retry = matches.get_flag(options::FOLLOW_RETRY);
let retry = arg_follow_retry || matches.get_flag(options::RETRY);
let follow = match (
    arg_follow_retry,
    matches
        .get_one::<String>(options::FOLLOW)
        .map(|s| s.as_str()),
) {
    (true, Some(_))
        if matches.index_of(options::FOLLOW_RETRY) > matches.index_of(options::FOLLOW) =>
    {
        Some(FollowMode::Name)
    }
    (_, Some("name")) | (true, None) => Some(FollowMode::Name),
    (_, Some(_)) => Some(FollowMode::Descriptor),
    (false, None) => None,
};
```
(I've omitted the comments here. Those are of course great to keep)

Reply:
Even more reduced (but the condition gets more complicated):
```rust
let arg_follow_retry = matches.get_flag(options::FOLLOW_RETRY);
let arg_follow = match matches
    .get_one::<String>(options::FOLLOW)
    .map(|s| s.as_str())
{
    Some("name") => Some(FollowMode::Name),
    Some(_) => Some(FollowMode::Descriptor),
    None => None,
};
let retry = arg_follow_retry || matches.get_flag(options::RETRY);
let follow = if arg_follow_retry
    && (arg_follow.is_none()
        || matches.index_of(options::FOLLOW_RETRY) > matches.index_of(options::FOLLOW))
{
    Some(FollowMode::Name)
} else {
    arg_follow
};
```